File: julia/basics/fibonacci.jl
# Print the first 20 Fibonacci numbers, filled into a preallocated vector.
# Run: julia fibonacci.jl

function fibonacci(count::Integer)
    values = Vector{Int}(undef, count)
    current, next = 0, 1
    for i in 1:count
        values[i] = current
        current, next = next, current + next
    end
    return values
end

println(join(fibonacci(20), " "))

File: r/basics/fibonacci.R
# Print the first 20 Fibonacci numbers, filled into a preallocated vector.
# Run: Rscript fibonacci.R

fibonacci <- function(count) {
  values <- integer(count)
  current <- 0L
  following <- 1L
  for (i in seq_len(count)) {
    values[i] <- current
    step <- current + following
    current <- following
    following <- step
  }
  values
}

cat(paste(fibonacci(20), collapse = " "), "\n", sep = "")
